crates/sui-framework/src/natives/object_runtime/mod.rs

```diff
@@ -396,19 +404,22 @@ impl ObjectRuntimeState {
         mut self,
         by_value_inputs: BTreeSet<ObjectID>,
         external_transfers: BTreeSet<ObjectID>,
-        mut loaded_child_objects: BTreeMap<ObjectID, SequenceNumber>,
+        loaded_child_objects: BTreeMap<ObjectID, SequenceNumber>,
         child_object_effects: BTreeMap<ObjectID, ChildObjectEffect>,
-        loaded_child_objects_fixed: bool,
+        _loaded_child_objects_fixed: bool,
     ) -> Result<RuntimeResults, ExecutionError> {
         let mut wrapped_children = BTreeSet::new();
-        if !loaded_child_objects_fixed {
-            loaded_child_objects = BTreeMap::new();
-            for (child, child_object_effect) in &child_object_effects {
-                if let Some(version) = child_object_effect.loaded_version {
-                    loaded_child_objects.insert(*child, version);
-                }
-            }
-        }
+
+        // This doesnt seem to make a difference and tests still pass. cc @tnowacki
+
+        // if !loaded_child_objects_fixed {
+        //     loaded_child_objects = BTreeMap::new();
+        //     for (child, child_object_effect) in &child_object_effects {
+        //         if let Some(version) = child_object_effect.loaded_version {
+        //             loaded_child_objects.insert(*child, version);
+        //         }
+        //     }
+        // }
```

Comment:
@tnowacki commenting this out did not make a difference. We absolutely sure it does anything?
The adapter tests passed too.

Reply:
We don't have tests that run on the old protocol config, but you will break sync with this gone

Reply:
I should clarify. I synced from genesis with this code commented out multiple times and no issues.